tests/Signals_perftest.m

```diff
@@ -1,90 +1,191 @@
+            end
+          end
+        case {'subscriptable'}
+          while (testCase.keepMeasuring)
+            for i = 1:testCase.Reps
+              post(testCase.A, s); % this post will trigger the update of signal 'c_A', defined above
+            end
+          end
       end
+      
     end
     
-    function test_benchmark(testCase)
-      x = testCase.A;
-      resfun = @(x)2.*x + x.*cos(x.*x - 5) + (x > 15) + 1;
-      y = resfun(x);
-      res = skipRepeats(delta(y));
+    function test_signalsOps(testCase, SignalsOps)
+      % @todo: create this test modeled off the above
+    end
+    
+    function test_networkOps(testCase, OnlyBasicOps, NumNodes, Depth)
+      % benchmarks operations on a Signals network upon posting to a signal
+      % which updates all other signals in the network
+      
+      minNodesPerLayer = floor(NumNodes/Depth);
+      % nodes in the 2nd layer of the network (add "leftover" nodes after
+      % even distribution of nodes across all other layers)
+      layer2Nodes = minNodesPerLayer + mod(NumNodes, Depth);
+      % array of number of nodes in each layer
+      nodesInLayers = iff(Depth > 1,... 
+        [layer2Nodes, minNodesPerLayer*ones(1, Depth-1)], layer2Nodes);
       
-      values = []; %zeros(1,nreps-1);
-      vidx = 1;
+      % set-up network to test
+      dependentSigs = cell(NumNodes, Depth); % cell array of all dependent signals
+      % number of ops to sample evenly
+      numOps = iff(OnlyBasicOps, length(testCase.BasicOps),... 
+        length(testCase.MainOps));
+      nodeNum = 1; % counter
       
-      list = res.onValue(@store);
-      res = res.buffer(4); %#ok<NASGU>
-      networkInfo(testCase.net.Id)
-      while(testCase.keepMeasuring)
-        for i = 1:testCase.reps
-          post(x, i);
+      for layer = 1:Depth
+        for curNode = 1:nodesInLayers(layer)
+          % get each op by equally sampling all ops
+          if OnlyBasicOps % can't use `iff` here b/c we are returning a function handle
+            op = testCase.BasicOps{mod(nodeNum, numOps)+1};
+          else
+            op = testCase.MainOps{mod(nodeNum, numOps)+1};
+          end
+          % create the dependent signal depending on the op
+          switch func2str(op)
+            
+            % for ops that do not create a new signal upon posting 1 to
+            % 'testCase.A', change op to '@identity'
+            case {'nop', 'post', 'subscriptable', 'onValue', 'identity'}
+              op = @identity;
+              dependentSigs{curNode, layer} = feval(op, testCase.A);
+            case {'gt', 'ge', 'lt', 'le', 'eq', 'plus', 'minus', 'times', 'rdivide'}
+              dependentSigs{curNode, layer} = feval(op, testCase.A, testCase.B);
+            case {'map'}
+              dependentSigs{curNode, layer} = feval(op, testCase.A, @(x) plus(1,x));
+            case {'scan'}
+              dependentSigs{curNode, layer} = feval(op, testCase.A, @plus, testCase.B);
```

Comment:
I think I'm missing something: if these always use A and B, then how do you achieve a depth > 1?

Reply:
This method is just for testing single operations. The width/depth testing is in the `test_networkOps` method